app/src/main/java/com/example/cinemapp/ui/main/ProfileFragment.kt

```diff
@@ -1,65 +1,184 @@
+                        viewModel.getNextPage()
+                    }
+                }
+            }
+        )
     }
 
-    private fun setupViews() {
+    private fun setupViews(accountDetails: AccountDetails) {
         with(binding) {
-            btnSignOut.visibility = View.INVISIBLE
-            btnSignOut.setOnClickListener {
-                lifecycleScope.launch {
-                    viewModel.userPrefs.deleteSessionId()
-                    startActivity(Intent(context, AuthenticationActivity::class.java))
+            tvName.text = accountDetails.name
+            tvUsername.text = accountDetails.username
+            if (accountDetails.avatar.isEmpty()) {
+                tvInitial.text = accountDetails.name.substring(0, 1)
+            } else {
+                    loadImage(
+                        accountDetails.avatar,
+                        ivAvatar,
+                        root.context,
+                        R.drawable.ic_placeholder_person
+                    )
+                    activity
+                    ivAvatar.setBackgroundColor(ContextCompat.getColor(requireContext(), R.color.md_theme_surface))
                 }
+            cpiLoading.visibility = View.INVISIBLE
+            clContent.visibility = View.VISIBLE
+        }
+    }
+
+    private fun setupOnClick() {
+        binding.btnSignOut.setOnClickListener {
+            lifecycleScope.launch {
+                viewModel.signOut()
             }
-            lifecycleScope.launch(Dispatchers.Main) {
-                viewModel.session.collect { sessionId ->
-                    tvUsername.text = sessionId ?: "Not signed in"
+        }
+        binding.btnSignIn.setOnClickListener {
+            startActivity(Intent(context, AuthenticationActivity::class.java))
+        }
+    }
 
-                    sessionId?.let {
-                        btnSignOut.visibility = View.VISIBLE
-                    }
+    private fun setupVisibility() {
+        lifecycleScope.launch(Dispatchers.Main) {
+            with(binding) {
+                btnSignIn.visibility = View.INVISIBLE
+                clContent.visibility = View.INVISIBLE
+                cpiLoading.visibility = View.VISIBLE
+            }
+        }
+
+    }
+
+
+    private fun setupTabs() {
+        with(binding){
+            tlMovieLists.selectTab(tlMovieLists.getTabAt(0))
+            tlMovieLists.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
+                override fun onTabSelected(tab: TabLayout.Tab?) {
+                    tab?.position?.let { onTabChanged(it) }
                 }
+
+                override fun onTabUnselected(tab: TabLayout.Tab?) {}
+
+                override fun onTabReselected(tab: TabLayout.Tab?) {}
+            })
+        }
+    }
+
+    private fun onTabChanged(tabPosition: Int) {
+        when (tabPosition) {
+            Tabs.FAVORITES.ordinal -> viewModel.getFavoriteNextPage()
```

Comment:
You can hide this fetching logic inside viewmodel by creating function called loadPage(tab: Tabs) and fetch a page you want